File: frontend/src/app/interfaces/quizDetail.ts
import {CategoryShort} from "./categoryShort";
import {QuestionShort} from "./questionShort";

export interface QuizDetail {
  id?: number;
  name: string;
  description: string;
  password?: string;
  isFavorite: boolean;
  category: CategoryShort;
  questions: QuestionShort[];
}

File: backend/src/main/java/org/yougrow/backend/entities/projection/Quiz/QuizDetail.java
package org.yougrow.backend.entities.projection.Quiz;

import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import org.yougrow.backend.entities.projection.IdAndName;
import org.yougrow.backend.entities.projection.IdAndQuestion;
import java.util.List;

@JsonPropertyOrder({"id", "name", "description", "isFavorite", "category", "questions"})
public interface QuizDetail {
    Long getId();

    String getName();

    String getDescription();

    Boolean getIsFavorite();

    IdAndName getCategory();

    List<IdAndQuestion> getQuestions();
}
